src/java/org/apache/cassandra/auth/AuthConfig.java

```diff
@@ -69,23 +70,16 @@ public static void applyAuth()
 
         // authorizer
 
-        IAuthorizer authorizer = new AllowAllAuthorizer();
-
-        if (conf.authorizer != null)
-            authorizer = FBUtilities.newAuthorizer(conf.authorizer);
+        IAuthorizer authorizer = authInstantiate(conf.authorizer, AllowAllAuthorizer.class);
 
         if (!authenticator.requireAuthentication() && authorizer.requireAuthorization())
-            throw new ConfigurationException(conf.authenticator + " can't be used with " + conf.authorizer, false);
+            throw new ConfigurationException(conf.authenticator.class_name + " can't be used with " + conf.authorizer, false);
 
         DatabaseDescriptor.setAuthorizer(authorizer);
 
         // role manager
 
-        IRoleManager roleManager;
-        if (conf.role_manager != null)
-            roleManager = FBUtilities.newRoleManager(conf.role_manager);
-        else
-            roleManager = new CassandraRoleManager();
+        IRoleManager roleManager = authInstantiate(conf.role_manager, CassandraRoleManager.class);
 
         if (authenticator instanceof PasswordAuthenticator && !(roleManager instanceof CassandraRoleManager))
```

Comment:
does not seem your change, but isn't it opposite - why am I missing here?

Reply:
Why do you think it should be the opposite? 

The way I read the intention of this code is that we shouldn't allow using `PasswordAuthenticator` or `MutualTlsWithPasswordFallbackAuthenticator` with any implementation of `IRoleManager` as `PasswordAuthenticator` has an implicit dependency on how passwords are created by `CassandraRoleManager`.

Is this wrong?

Reply:
According to our slack discussion, I've made a change in in this this message.